packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -206,23 +206,39 @@ export class OdspDocumentService implements IDocumentService {
                 ? Promise.resolve(null)
                 : this.getWebsocketToken!(options);
             const joinSessionPromise = this.joinSession(requestWebsocketTokenFromJoinSession).catch((e) => {
-                let code: string | undefined;
+                const stack: any = [];
                 try {
-                    code = e?.response ? JSON.parse(e?.response)?.error?.code : undefined;
+                    if(e !== undefined) {
+                        stack.push(e.code);
+                        if(e.response !== undefined) {
+                            stack.push(e.response.code);
+                            if(e.response.error !== undefined) {
```

Comment:
I believe it's e.response.error.[innerError...].code, i.e. recursively, with infinite level of recursion (handled by code).
I.e. 
e.response.error.code
e.response.error.innerError.code
e.response.error.innerError.innerError.code
e.response.error.innerError.innerError.innerError.code
...